Add time to brewing message

KlaasH opened the pull request:
The status channel being mostly the same exact message makes my eyes
glaze over a bit when I check it, plus it takes a tiny amount of mental
effort to read the little timestamp and add five minutes to it. So this
changes the brewing message to say what time it will be ready.

KlaasH committed:
Add time to brewing message

The status channel being mostly the same exact message makes my eyes
glaze over a bit when I check it, plus it takes a tiny amount of mental
effort to read the little timestamp and add five minutes to it. So this
changes the brewing message to say what time it will be ready.

hectcastro commented on functions/Slack/main.py:
```diff
@@ -27,8 +29,10 @@ def handle(event, context):
+        ready_time = now + datetime.timedelta(minutes=5)
         requests.post(slack_webhook_url, json={
-            'text': 'Coffee is brewing and will be ready in about 5 minutes!',
+            'text': 'Coffee is brewing and will be ready at {}!'.format(ready_time.strftime('%-I:%M%p')),  # NOQA
```
When I test this locally, I get:

```
Coffee is brewing and will be ready at !
```

If I convert the `%-I` to `%I`, I get:

```
Coffee is brewing and will be ready at 07:26PM!
```

KlaasH replied:
Hmm.  "The full set of format codes supported varies across platforms, because Python calls the platform C library’s strftime() function, and platform variations are common." (https://docs.python.org/2/library/datetime.html#strftime-and-strptime-behavior)

Are you on a Mac?  I guess even if Lambda supports `%-I` it's probably worth doing the more-compatible thing.

hectcastro replied:
I was, but I was testing within the context of a Linux container (the image definition included here). Should be Debian Jessie. Unclear what's going on.

KlaasH committed:
Tweak time format string

Switches from `%-I` to `%l` (that's a small L) for compatibility and
lowercases the AM/PM.

hectcastro commented:
👍

KlaasH commented:
One could wish that Python just implemented strptime and didn't depend on the murky pool of the platform implementation at all.

I used `%-I` based on [strftime.org](http://strftime.org/), which calls itself "Python strptime reference", and because it worked.  However the [actual Python documentation](https://docs.python.org/2/library/datetime.html#strftime-and-strptime-behavior) makes no mention of it.  Nor do either of those mention `%l` (that's a lower-case L), which is documented in the [OS X version](https://developer.apple.com/library/ios/documentation/System/Conceptual/ManPages_iPhoneOS/man3/strftime.3.html) and the man page on my system (Trusty).  It differs from `%-I` in that it includes a leading zero for single-digit times, but that's easily dealt with.

So I changed the second commit to use `%l` to get rid of the leading zero in what I hope is a reasonably cross-platform way...

KlaasH commented:
Builds are failing because the good folks at Travis [accidentally blowed up their build environment images](https://www.traviscistatus.com/incidents/2p40l49r3yxd).  (The error is `make: docker: Command not found`).
